### PR TITLE
[Docs] Add catalog of labels to README

### LINKED ISSUE #846: Add section to README with catalog of device plugin specific labels
GFD docs here have a nice list [here](https://github.com/NVIDIA/k8s-device-plugin/tree/main/docs/gpu-feature-discovery#generated-labels) of the labels written by GFD. It would be great to have something similar for the device plugin itself, how they're used, and in which configurations they are changed. Or, in other cases, when device plugin will modify (and how) labels set by GFD.

Ex., `nvidia.com/device-plugin.config` is not written by but read by the device plugin. `nvidia.com/gpu.replicas` is written by and modified by the device plugin according to a sharing config. `nvidia.com/gpu.sharing-strategy` is owned by the device plugin. `nvidia.com/gpu.product` is written by GFD but may be modified by the device plugin. `nvidia.com/mig.capable`, `nvidia.com/mps.capable`, and `nvidia.com/vgpu.present` all look like they're owned by the device plugin.

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: chipzoller <chipzoller@gmail.com>

Closes #846

This PR adds a new "catalog of labels" section to the device plugin's README and updates the same table for GFD. It also makes a few minor improvements:
* Lints/standardizes common Markdown elements
* Improves code blocks
* Fixes a couple things
* Updates version numbers to reflect current (v0.16.2)

### CODE REVIEW

**Comment by chipzoller** [comment]
Could I get a review from you please, @mikemckiernan?

File: README.md
Comment (mikemckiernan) [inline README.md:603]: Ditto above, need SME review.

```suggestion
| nvidia.com/mps.capable            | Boolean    | Specifies if devices on the node are configured for MPS.                                                                                                                                                  | false          |
```

File: README.md
Comment (mikemckiernan) [inline README.md:604]:
```suggestion
| nvidia.com/vgpu.present           | Boolean    | Specifies if devices on the node use vGPU.                                                                                                                                                                     | false          |
```

File: README.md
Comment (mikemckiernan) [inline README.md:605]:
```suggestion
| nvidia.com/vgpu.host-driver-version | String   | Specifies the vGPU host driver version on the underlying hypervisor.                                                                                                                   | 550.54.16       |
```

File: README.md
Comment (mikemckiernan) [inline README.md:606]:
```suggestion
| nvidia.com/vgpu.host-driver-branch | String    | Specifies the vGPU host driver branch on the underlying hypervisor.                                                                                                                    | r550_40           |
```

**Review by mikemckiernan — COMMENTED** [review]
Thank you again for the assist!  PLMK what I can clarify.

**Comment by chipzoller** [comment]
Thank you for your review, @mikemckiernan. Could we get technical review here first so we ensure the content is accurate before we do any word smithing?

**Comment by chipzoller** [comment]
Requesting technical review, please.

**Comment by chlunde** [comment]
@chipzoller Could you also document if it is possible to not deploy GFD, and the minimum set of labels you would have to apply manually (for example using Karpenter)?

**Comment by chipzoller** [comment]
The docs [here](https://github.com/NVIDIA/k8s-device-plugin?tab=readme-ov-file#deploying-with-gpu-feature-discovery-for-automatic-node-labels) already cover opting in to deploying GFD. As for the other part, not sure I follow.

**Comment by chipzoller** [comment]
Would one of the project maintainers please provide technical review or, alternatively, close this PR if it is undesirable? cc @cdesiniotis, @tariq1890, and @elezar.

**Comment by chipzoller** [comment]
Ok, @mikemckiernan, I can't seem to get anyone else to review this PR so I've gone ahead and implemented all your suggestions. Maybe you can take a look at this again and possibly ping someone on your side? Or, again, if this is just unwanted work go ahead and close.

**Comment by chipzoller** [comment]
@klueska, any way you can give this a quick read for technical accuracy?

**Comment by mikemckiernan** [comment]
@chipzoller, I appreciate the assist and I'm sure my colleagues do too.  Time is just so tight.  Thanks for your help and patience.

**Review by mikemckiernan — APPROVED** [review]

**Comment by tariq1890** [comment]
Hi @chipzoller , thanks for your contribution here! The changes LGTM, can you squash your commit history ?

**Comment by chipzoller** [comment]
Thanks for the review, @tariq1890. I'm having a difficult time squashing commit history here because it's now so long due to the merges from `main`. I don't understand though, why can't you all just squash upon PR merge so you're left with just one commit? What's the functional difference here?

**Comment by tariq1890** [comment]
> why can't you all just squash upon PR merge so you're left with just one commit? What's the functional difference here?

I can't speak for others, but it has always been my practice to squash the commit history myself rather than have GitHub do it. This way, we enforce a practice where developers take more control over the git history they will generate for the repo instead of having GitHub "intervene". Moreover, we are only much more like to hit `Merge Pull Request` button, instead of actually clicking on the drop down next to the button and specifying Squash. For these two reasons, my preference is that the contributor squash their commit history themselves

I've enabled squash ing of commits, so we're good for now. Thanks again

**Pull request merged**
